tests/network/test_add_intercept.py

```diff
@@ -255,3 +257,87 @@ async def test_add_intercept_type_pattern_port_empty_invalid(websocket):
         "error": "invalid argument",
         "message": "URL pattern must specify a port"
     } == exception_info.value.args[0]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "url_patterns",
+    [
+        [
+            {
+                "type": "string",
+                "pattern": "https://www.example.com/",
+            },
+        ],
+        # [
+        #     {
+        #         "type": "pattern",
+        #         "protocol": "https",
+        #         "hostname": "www.example.com",
+        #         "pathname": "/",
+        #     },
+        # ],
+        # [
+        #     {
+        #         "type": "string",
+        #         "pattern": "https://www.example.com/",
+        #     },
+        #     {
+        #         "type": "pattern",
+        #         "protocol": "https",
+        #         "hostname": "www.example.com",
+        #         "pathname": "/",
+        #     },
+        # ],
+    ],
+    ids=[
+        "string",
+        #  "pattern",
+        #  "string and pattern",
+    ])
+async def test_add_intercept_blocks(websocket, context_id, url_patterns):
+    url = "https://www.example.com/"
```

Comment:
do we want to test with example.com or perhaps set up a local server?

Reply:
We still do not have local server support: https://github.com/GoogleChromeLabs/chromium-bidi/issues/575

Reply:
should we address it next? we will need to have many more tests were the example.com might not be sufficient or reliable.

Reply:
I agree, the more the time passes the more important this feature becomes. Let's try to prioritize it. There are some pre-existing tests that use example.com that would also benefit from local servers.

(Out of scope of this PR though.)